README.md
<!--
  Title: CONSUL
  Description: Citizen Participation and Open Government Application
  Keywords: democracy, citizen participation, eparticipation, debates, proposals, voting, consultations, crowdlaw, participatory budgeting
-->

![CONSUL logo](https://raw.githubusercontent.com/consul/consul/master/public/consul_logo.png)

# CONSUL

Citizen Participation and Open Government Application - City of Torino general version

[![Build Status](https://travis-ci.org/consul/consul.svg?branch=master)](https://travis-ci.org/consul/consul)
[![Code Climate](https://codeclimate.com/github/consul/consul/badges/gpa.svg)](https://codeclimate.com/github/consul/consul)
[![Coverage Status](https://coveralls.io/repos/github/consul/consul/badge.svg)](https://coveralls.io/github/consul/consul?branch=master)
[![Crowdin](https://d322cqt584bo4o.cloudfront.net/consul/localized.svg)](https://crowdin.com/project/consul)
[![License: AGPL v3](https://img.shields.io/badge/License-AGPL%20v3-blue.svg)](http://www.gnu.org/licenses/agpl-3.0)
[![Reviewed by Hound](https://img.shields.io/badge/Reviewed_by-Hound-8E64B0.svg)](https://houndci.com)

[![Accessibility conformance](https://img.shields.io/badge/accessibility-WAI:AA-green.svg)](https://www.w3.org/WAI/eval/Overview)
[![A11y issues checked with Rocket Validator](https://rocketvalidator.com/badges/checked_with_rocket_validator.svg?url=https://rocketvalidator.com)](https://rocketvalidator.com/opensource)

[![Join the chat at https://gitter.im/consul/consul](https://badges.gitter.im/consul/consul.svg)](https://gitter.im/consul/consul?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
[![PRs Welcome](https://img.shields.io/badge/PRs-welcome-brightgreen.svg?style=flat-square)](https://github.com/consul/consul/issues?q=is%3Aissue+is%3Aopen+label%3APRs-welcome)

This is the opensource code repository of the eParticipation website CONSUL, originally developed for the Madrid City government eParticipation website

## Documentation

Check the ongoing documentation at [https://docs.consulproject.org](https://docs.consulproject.org) to learn more about how to start your own CONSUL fork, install it, customize it and learn to use it from an administrator/maintainer perspective.

## CONSUL Project main website

You can access the main website of the project at [http://consulproject.org](http://consulproject.org) where you can find documentation about the use of the platform, videos, and links to the community space.

## Configuration for development and test environments

**NOTE**: For more detailed instructions check the [docs](https://docs.consulproject.org)

Prerequisites: install git, Ruby 2.3.2, `bundler` gem, Node.js and PostgreSQL (>=9.4).

```bash
git clone https://github.com/consul/consul.git
cd consul
bundle install
cp config/database.yml.example config/database.yml
cp config/secrets.yml.example config/secrets.yml
bin/rake db:create
bin/rake db:migrate
bin/rake db:csi_seed
RAILS_ENV=test rake db:setup
```

Run the app locally:

```bash
bin/rails s
```

Prerequisites for testing: install ChromeDriver >= 2.33

Run the tests with:

```bash
bin/rspec
```

You can use the default admin user from the seeds file:

 **user:** admin@consul.dev
 **pass:** 12345678

But for some actions like voting, you will need a verified user, the seeds file also includes one:

 **user:** verified@consul.dev
 **pass:** 12345678

## Configuration for production environments

See [installer](https://github.com/consul/installer)

## Current state

Development started on [2015 July 15th](https://github.com/consul/consul/commit/8db36308379accd44b5de4f680a54c41a0cc6fc6). Code was deployed to production on 2015 september 7th to [decide.madrid.es](https://decide.madrid.es). Since then new features are added often. You can take a look at the current features at the [project's website](http://consulproject.org/) and future features at the [Roadmap](https://github.com/consul/consul/projects/6) and [open issues list](https://github.com/consul/consul/issues).

## License

Code published under AFFERO GPL v3 (see [LICENSE-AGPLv3.txt](LICENSE-AGPLv3.txt))

## Contributions

See [CONTRIBUTING.md](CONTRIBUTING.md)


CHANGELOG.md
# Changelog
All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).

## [1.4.2](https://github.com/csipiemonte/consul/compare/v1.4.1...v1.4.2) - 2019-09-30

### Changed
- Upd Gemfile
- Fixed logout SPID session (staging)

## [1.4.1](https://github.com/csipiemonte/consul/compare/v1.4.0...v1.4.1) - 2019-07-31

### Added

- **Admin:** New settings `feature.remote_census`
## [1.4.0](https://github.com/csipiemonte/consul/compare/v1.3.3...v1.4.0) - 2019-07-26

### Added
- **Admin:** Add document uploads from admin sectio
- **Admin:** Improve Admin settings section
- **Admin:** Add download's dashboard
- **Dashboard:** Add proposal's dashboard
- **Milestones:** Add progress bars to milestones public view

### Changed
- Privacy and census_terms moved to DB

## [1.3.3](https://github.com/csipiemonte/consul/compare/v1.3.2...v1.3.3) - 2018-12-14

### Added
- New settings `hot_score_period_in_days`, `feature.featured_proposals`
- Enhancements to admin backoffice

## [1.3.2](https://github.com/csipiemonte/consul/compare/v1.3.1...v1.3.2) - 2018-12-04

### Changed
- Privacy and conditions pages

## [1.3.1](https://github.com/csipiemonte/consul/compare/v1.3.0...v1.3.1) - 2018-11-16

### Added
- New setting `feature.help_page`
- Node.js to compile assets

### Removed
- Proposal guide

## [1.3.0](https://github.com/csipiemonte/consul/compare/v1.2.5...v1.3.0) - 2018-08-31

### Added
- Homepage: cards, header and cards seeds; customization of homepage from admin section
- Recommendations: Debates and proposals recommendations for users
- Configuration: Added setting on admin to skip user verification
- Admin newsletter emails
- New setting `feature.allow_images` to allow upload and show images for both (proposals and budget investment projects)
- Related Content feature. Now Debates & Proposals can be related

## [1.2.5](https://github.com/csipiemonte/consul/compare/v1.2.4...v1.2.5) - 2018-07-09

### Changed
- Votes to legislation proposals not only for verified users

## [1.2.4](https://github.com/csipiemonte/consul/compare/v1.2.3...v1.2.4) - 2018-07-02

### Changed
- Summary of Subway line 2 card 

## [1.2.3](https://github.com/csipiemonte/consul/compare/v1.2.2...v1.2.3) - 2018-07-02

### Changed
- Subway line 2 in home page
- Comments to legislation processes not only for verified users
- Footer

## [1.2.2](https://github.com/csipiemonte/consul/compare/v1.2.1...v1.2.2) - 2018-04-12

### Changed
- Logout SPID sessions

## [1.2.1](https://github.com/csipiemonte/consul/compare/v1.2.0...v1.2.1) - 2018-02-13

### Changed
- Home, FAQ, conditions, more info page

## [1.2.0](https://github.com/csipiemonte/consul/compare/v1.1.0...v1.2.0) - 2018-02-09

### Changed
- Home page slots
- Labels text
- FAQ, more info page
- Delayed Job disabled for production env

## [1.1.0](https://github.com/csipiemonte/consul/compare/v1.0.1...v1.1.0) - 2018-01-23

### Added
- SPID authentication
- Symlink to data/public/system (docs and images)
- Symlink to data/tmp

### Changed
- Log file path

## [1.0.1](https://github.com/csipiemonte/consul/compare/v1.0.0...v1.0.1) - 2017-12-05

### Changed
- Welcome page CSS

## [1.0.0](https://github.com/csipiemonte/consul/compare/v0.10...v1.0.0) - 2017-12-04

CSI Piemonte initial version.

## [0.10](https://github.com/consul/consul/compare/v0.9...v0.10) - 2017-07-05
### Added
- Milestones on Budget Investment's
- Feature flag to enable/disable Legislative Processes
- Locale site pages customization
- Incompatible investments

### Changed
- Localization files reorganization. Check migration instruction at https://github.com/consul/consul/releases/tag/v0.10
- Rails 4.2.9

## [0.9.0](https://github.com/consul/consul/compare/v0.8...v0.9) - 2017-06-15
### Added
- Budgets
- Basic polls
- Collaborative legistlation
- Custom pages
- GraphQL API
- Improved admin section

### Changed
- Improved admin section
- Rails 4.2.8
- Ruby 2.3.2

### Deprecated
- SpendingProposals are deprecated now in favor of Budgets

### Fixed
- CKEditor locale compilation fixed
- Fixed bugs in mobile layouts

## [0.8.0](https://github.com/consul/consul/compare/v0.7...v0.8)- 2016-07-21
### Added
- Support for customization schema, vía specific custom files, assets and folders

### Changed
- Rails 4.2.7
- Ruby 2.3.1

### Fixed
- Fixed bug causing errors on user deletion

## [0.7.0] - 2016-04-25
### Added
- Debates
- Proposals
- Basic Spending Proposals

### Changed
- Rails 4.2.6
- Ruby 2.2.3

[Unreleased]: https://github.com/consul/consul/compare/v0.10...consul:master
[0.10.0]: https://github.com/consul/consul/compare/v0.9...v0.10
[0.9.0]: https://github.com/consul/consul/compare/v0.8...v0.9
[0.8.0]: https://github.com/consul/consul/compare/v0.7...v0.8


db/csi_seeds.rb
# require "database_cleaner"
# DatabaseCleaner.clean_with :truncation

# coding: utf-8
# Default admin user (change password after first deploy to a server!)
if Administrator.count == 0
  admin = User.create!(username: "DecidiTorino", email: "admin@deciditorino.it", password: "12345678",
  	                   password_confirmation: "12345678", confirmed_at: Time.current,
  	                   terms_of_service: "1")
  admin.create_administrator
end

Setting.reset_defaults

WebSection.where(name: "homepage").first_or_create
WebSection.where(name: "debates").first_or_create
WebSection.where(name: "proposals").first_or_create
WebSection.where(name: "budgets").first_or_create
WebSection.where(name: "help_page").first_or_create

# Default custom pages
load Rails.root.join("db", "pages.rb")

puts " ✅"
print "Creating Geozones (Circoscrizioni)"

Geozone.create(name: "Centro - Crocetta",
	           external_code: "001", census_code: "01",
	           html_map_coordinates: "159, 132, 134, 183, 136, 190, 134, 195, 151, 206, 172, 169, 188, 178, 199, 165, 188, 148, 177, 146, 159, 132")
Geozone.create(name: "Santa Rita - Mirafiori Nord - Mirafiori Sud",
	           external_code: "002", census_code: "02",
	           html_map_coordinates: "89, 180, 117, 188, 127, 187, 134, 189, 131, 196, 143, 203, 101, 254, 122, 266, 126, 256, 138, 262, 138, 266, 132, 284, 126, 282, 115, 285, 107, 281, 105, 280, 94, 276, 91, 283, 86, 282, 74, 276, 65, 277, 58, 271, 39, 266, 40, 258, 52, 261, 63, 235, 53, 238, 23, 219, 19, 211, 21, 209, 27, 213, 31, 207, 70, 232, 73, 218, 77, 219, 80, 208, 84, 202, 79, 200, 80, 198, 87, 190, 89, 181")
Geozone.create(name: "San Paolo - Cenisia - Pozzo Strada - Cit Turin - Borgata Lesna",
	           external_code: "003", census_code: "03",
	           html_map_coordinates: "62, 144, 152, 142, 137, 187, 125, 185, 115, 187, 90, 179, 89, 173, 76, 175, 70, 173, 65, 171, 66, 164, 74, 162, 73, 158, 67, 156, 66, 150, 59, 149, 61, 144")
Geozone.create(name: "San Donato - Campidoglio - Parella",
	           external_code: "004", census_code: "04",
	           html_map_coordinates: "82, 99, 80, 110, 75, 115, 72, 116, 69, 123, 62, 143, 154, 139, 167, 112, 133, 109, 131, 119, 75, 90, 82, 99")
Geozone.create(name: "Borgo Vittoria - Madonna di Campagna - Lucento - Vallette",
	           external_code: "005", census_code: "05",
	           html_map_coordinates: "92, 62, 76, 89, 126, 115, 136, 105, 168, 111, 200, 61, 144, 52, 130, 68, 123, 67")
Geozone.create(name: "Barriera di Milano - Regio Parco - Barca - Bertolla - Falchera - Rebaudengo - Villaretto",
	           external_code: "006", census_code: "06",
	           html_map_coordinates: "136, 38, 145, 48, 165, 41, 172, 55, 183, 52, 192, 61, 198, 61, 200, 70, 172, 111, 187, 113, 203, 132, 231, 115, 234, 103, 248, 104, 262, 108, 281, 89, 258, 73, 254, 44, 232, 40, 227, 33, 237, 17, 225, 14, 223, 23, 197, 28, 171, 13, 155, 12, 148, 20, 138, 18, 143, 30")
Geozone.create(name: "Aurora - Vanchiglia - Sassi - Madonna del Pilone",
	           external_code: "007", census_code: "07",
	           html_map_coordinates: "242, 106, 268, 114, 282, 131, 310, 141, 251, 221, 242, 219, 236, 210, 236, 187, 217, 158, 200, 163, 187, 144, 159, 130, 171, 112, 188, 116, 203, 136, 233, 115, 240, 107")
Geozone.create(name: "San Salvario - Cavoretto - Borgo Po - Nizza Millefonti - Lingotto - Filadelfia",
	           external_code: "008", census_code: "08",
	           html_map_coordinates: "172, 172, 190, 180, 212, 161, 234, 190, 232, 213, 240, 223, 240, 225, 233, 236, 195, 253, 178, 248, 166, 266, 139, 265, 137, 259, 124, 254, 121, 263, 104, 253, 144, 205, 152, 210, 172, 170")

puts " ✅"
print "Creating Tags Categories"

ActsAsTaggableOn::Tag.category.create!(name: I18n.t("seeds.categories.associations"))
ActsAsTaggableOn::Tag.category.create!(name: I18n.t("seeds.categories.culture"))
ActsAsTaggableOn::Tag.category.create!(name: I18n.t("seeds.categories.sports"))
ActsAsTaggableOn::Tag.category.create!(name: I18n.t("seeds.categories.social_rights"))
ActsAsTaggableOn::Tag.category.create!(name: I18n.t("seeds.categories.economy"))
ActsAsTaggableOn::Tag.category.create!(name: I18n.t("seeds.categories.employment"))
ActsAsTaggableOn::Tag.category.create!(name: I18n.t("seeds.categories.equity"))
ActsAsTaggableOn::Tag.category.create!(name: I18n.t("seeds.categories.sustainability"))
ActsAsTaggableOn::Tag.category.create!(name: I18n.t("seeds.categories.participation"))
ActsAsTaggableOn::Tag.category.create!(name: I18n.t("seeds.categories.mobility"))
ActsAsTaggableOn::Tag.category.create!(name: I18n.t("seeds.categories.media"))
ActsAsTaggableOn::Tag.category.create!(name: I18n.t("seeds.categories.health"))
ActsAsTaggableOn::Tag.category.create!(name: I18n.t("seeds.categories.transparency"))
ActsAsTaggableOn::Tag.category.create!(name: I18n.t("seeds.categories.security_emergencies"))
ActsAsTaggableOn::Tag.category.create!(name: I18n.t("seeds.categories.environment"))

puts " ✅"
puts "All CSI seeds created successfuly 👍"
